futures-util/src/lock/mod.rs

```diff
@@ -0,0 +1,10 @@
+//! Futures-powered synchronization primitives.
```

Comment:
Should this be in futures-util/lock, or could there be a crate like `futures-sync[-primitives]`.

The reason for this is
- channels have their own crate
- channels might be rewritten or variants added that might make use of those more basic asynchronous sync primitives
- a Mutex only only one variant of an asynchronous sync primitive. Semaphores and Events are others (and those are not necessarily locks)

Reply:
I don't have a strong opinion here, but I'd like to avoid introducing another crate now. I think this makes sense to expose through `futures-util` since futures-util is sort of the "dumping ground" for new ideas that we're not sure we want to stabilize on, and which aren't ready for people to expose through public APIs. Channels are intended to be something with a more stable API that can appear in the public API of libraries, which is why they have their own crate to avoid a semver bump when something in futures-util gets a breaking change.